sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs

```diff
@@ -366,6 +386,10 @@ private Instruction CreateInstruction(SyntaxNode syntaxNode, string methodId, st
             private ISymbol GetSymbol(SyntaxNode syntaxNode) =>
                 semanticModel.GetSymbolInfo(syntaxNode).Symbol;
 
+            private static bool AcceptsOrReturnsString(IMethodSymbol methodSymbol) =>
+                methodSymbol.ReturnType.Is(KnownType.System_String) ||
+                methodSymbol.Parameters.Any(p => p.Type.Is(KnownType.System_String));
```

Comment:
I am not sure this test will be correct for EF Core. Look here https://github.com/aspnet/EntityFrameworkCore/blob/2942b6035887e97e316b018f518a8150497465b3/src/EFCore.Relational/RelationalDatabaseFacadeExtensions.cs#L130 they expect a `RawSqlString` but maybe we shouldn't care as most of the time it will be a string passed on the call site and not an actual instance of `RawSqlString`

Reply:
See [this line](https://github.com/SonarSource/sonar-csharp/pull/1406/files#diff-8ca0cb6e76db90b64f9fa80af5c40a6fR283), it should allow methods that receive string arguments but their parameters are not string (usually object, but should work with RawSqlString too). Of course for the time being it will not work if you construct RawSqlString.